particles/Tutorial/Kotlin/4_Handles/DisplayGreeting.kt

```diff
@@ -0,0 +1,32 @@
+package arcs.tutorials
+
+import arcs.*
+import kotlin.native.internal.ExportForCppRuntime
+
+/**
+ * Sample WASM Particle.
+ */
+class DisplayGreetingParticle : Particle() {
+    private val person = Singleton { DisplayGreeting_Person() }
+
+    override fun getTemplate(slotName: String) = "Hello, <span>{{name}}</span>!"
+
+     init {
+         registerHandle("person", person)
+     }
+
+    override fun onHandleUpdate(handle: Handle) {
+         this.renderOutput()
+    }
+
+    override fun populateModel(slotName: String, model: Map<String, Any?>): Map<String, Any?> {
+        val n = person.get()!!.name ?: "Human"
```

Comment:
if `person.get()` is nullable, you can drop the double-bang and switch to a cleaner:

```kotlin
val n = person.get()?.name ?: "Human"
```